src/app/base/components/bricks-integrator/bricks-integrator.component.ts

```diff
@@ -340,5 +346,13 @@ export class BricksIntegratorComponent implements OnInit, OnDestroy {
     if (this.newTaskId.observers.length > 0) this.newTaskId.emit(taskId);
   }
 
-
+  checkIfFunctionNameIsUnique(name: string) {
+    this.codeParser.nameTaken = !!(this.nameLookups?.find(x => x == name));
+    name = toPythonFunctionName(name);
+    if (this.config.preparedCode) {
+      this.codeParser.functionName = name;
+      this.config.preparedCode = this.config.preparedCode.replace(new RegExp(getPythonFunctionName(this.config.preparedCode), 'g'), name);
```

Comment:
this wasn't meant with the previous comment. I mean that a simple replace of the name might run into issues independent of global or not (even more so with global but in general not viable).  
![image](https://user-images.githubusercontent.com/91723236/216298033-2e9724c5-8339-4b69-94e1-ae4db679cbb1.png)


Also this doesn't really belong in the component ts file. 
I'd go with a function in the codeparser that is used to set the function name that in turn handles the preparedCode. This should also ensure that we always start from scratch in the preparation. I want to remove the possibility that someone changes a function name and changes it again and arrives at a different result than selecting the final name from the beginning.

Also the function name might be misleading since it's more of a checkFunctionNameAndSet or something similar since it not only to check